src/zkevm_specs/evm/execution/end_tx.py

```diff
@@ -33,6 +34,28 @@ def end_tx(instruction: Instruction):
     coinbase = instruction.block_context_lookup(BlockContextFieldTag.Coinbase)
     instruction.add_balance(coinbase, [reward])
 
+    # constrain tx status matches with `PostStateOrStatus` of TxReceipt tag in RW
+    instruction.constrain_equal(
+        is_persistent, instruction.tx_receipt_lookup(tx_id, TxReceiptFieldTag.PostStateOrStatus)
+    )
+
+    # constrain log id matches with `LogLength` of TxReceipt tag in RW
+    log_id = instruction.tx_receipt_lookup(tx_id, TxReceiptFieldTag.LogLength)
+    instruction.constrain_equal(log_id, instruction.curr.log_id)
+
+    # constrain `CumulativeGasUsed` of TxReceipt tag in RW
+    if tx_id == 1:  # check if it is the first tx
+        pre_tx_gas_cumulate = FQ(0)
+    else:
+        pre_tx_gas_cumulate = instruction.tx_receipt_lookup(
+            tx_id - FQ(1), TxReceiptFieldTag.CumulativeGasUsed
+        ).expr()
+
+    instruction.constrain_equal(
+        pre_tx_gas_cumulate + gas_used,
+        instruction.tx_receipt_lookup(tx_id, TxReceiptFieldTag.CumulativeGasUsed),
+    )
+
```

Comment:
```suggestion
    instruction.constrain_equal(
        current_cumulative_gas_used + gas_used,
        instruction.tx_receipt_lookup(tx_id, TxReceiptFieldTag.CumulativeGasUsed),
    )

```

Reply:
updated in https://github.com/appliedzkp/zkevm-specs/pull/181/commits/c3c7c976ba873a4ccff611f12d3309f6d20622e3 thanks